### DISCUSSION TITLE
[FB7490] kernel Panic direkt nach dem Flashen: Unable to handle kernel paging

### DISCUSSION THREAD

**Discussion by SvenLuebke** [Freetz-NG/freetz-ng#967] [Allgemein]
Hallo!

Ich habe mich entschlossen freetz-ng auf einer 7490 zu testen (commit 943a6f9). Hab alles ohne Probleme auf Basis des OS 7.57 durchgebaut bekommen, das Image erstellt und anschließend über EVA-FTP-Client.ps1 (in-memory File) geflasht. Allerdings tritt relativ schnell nach dem Starten eine Kernel Panic auf. Was kann ich dagegen tun? Hat die FritzBox nicht genug RAM? Ein altes 2019er freetz-Image, das ich im Internet gefunden habe, scheint zu laufen.

Danke für Eure Hilfe!

[7490_freetz_kernelPanic.txt](https://github.com/Freetz-NG/freetz-ng/files/14833189/7490_freetz_kernelPanic.txt)

**Comment by fda77** [comment]
> [_AVM_WATCHDOG_ctrl_timer]Warning! last hw-trigger before 7 s (WDT_DEFAULT_TIME 5 s)

Du kannst als "DEVELOPER" einen Patch aktivieren um den AVM-Watchdog zu deaktivieren. Damit hast du mehr Zeit zu schauen was nicht geht. Die Ursache wird dadurch nicht behoben!
Hast du komisch Kernelmodule gebaut und geladen? "modprobe" hat die höchste PID. Dieses "D3N_RessourceInit" ist wohl irgend was von AVM mit ISDN. Hast du die ganzen Telefonfunktionen genutzt? Ging die Hardware noch? Älteres FOS soll wohl HW-Defekte ignoriert haben, neuere rebooten.
Ansonsten zum testen keine replace-kernel, keine module, keine remove-patches usw, wie immer. Auf meiner 7490 läuft es jedenfalls

**Reply by SvenLuebke** [reply]
Danke, dass man den Watchdog darüber deaktivieren kann, wusste ich noch nicht. Ich hatte in der Tat auch ein paar wenige Dinge herauspatchen lassen, um Platz zu sparen, allerdings den original Kernel verwendet. Nun habe ich nach bestem Wissen und Gewissen alles auf Originalzustand gebracht (abgesehen von zusätzlichen Software-Packages) und das kdsldmod kernel module scheint zu crashen (obwohl es in dem anderen Build nicht gecrasht ist).

Komische Kernelmodule hab ich keine gebaut oder geladen. Die "ganzen Telefonfunktionen" würde ich schon gerne nutzen, wie in der original Firmware. Die Fritzbox hat sich flashen lassen, dann hat es ungefähr eine Minute gedauert (also "........" im ADAM2, Kernel fährt bis zu einem bestimmten Punkt) und dann CRASH, Kernel Panic, diverse Backtraces und dann
```
[   64.290000][1]Kernel panic - not syncing: Fatal exception 
[   64.290000] Rebooting in 5 seconds..
```

Die Hardware bzw. besser Freetz konnte ich also nicht nutzen. Wenn ich die original 7.57 Firmware flashe, bekomme ich keine Fehler. Kann man sich irgendwie fertige Images herunterladen? Oder könntest Du mir ein Image zum Testen zur Verfügung stellen?

**Comment by fda77** [comment]
> Die "ganzen Telefonfunktionen" würde ich schon gerne nutzen, 

Die Frage ist mehr ob die Hardware dafür okay ist, neues FOS sind intoleranter als ältere. Dieses D3N_RessourceInit hat damit zu tun.
Bau halt zuerst ein minimal-image um herauszufinden woher das problem kommt, sonst hab ich keine Idee
... abgesehen mal in rc.custom etc Altlasten aufzuräumen,

**Reply by SvenLuebke** [reply]
Ich gehe davon aus, dass Du mit "FOS" das FritzOS meinst? Mag ja sein, dass die neueren Versionen intoleranter sind, aber warum scheint das original Image der Version 7.57 (von AVM) keine Probleme zu machen? Gleicher Kernel, gleiches kdsldmod. Klar, das Image ist etwas kleiner. Dann könnte es höchstens sein, dass das Flash in den hinteren Bereichen defekt ist. Oder die Image-Daten werden nicht vernünftig übertragen, aber warum funktioniert es dann im original Image? Oder der Fehler tritt durch irgendwelche Randbedingungen erst nach längerer Nutzungsdauer auf.

Eine Erklärung finde ich für meine Probleme gerade nicht. Leider hab ich vor zwei Monaten alle anderen 7490 verkauft, ansonsten hätte ich noch andere testen können.

**Reply by SvenLuebke** [reply]
Ich habe nun sicherheitshalber ein anderes Netzteil verwendet und viele removal Patches aktiviert, wodurch die Firmware knapp über 43MB groß war: kdsld sowie dect und weiteres habe "ich" entfernt. Das Image läuft nun. Der Flashvorgang läuft nun auch etwas anders ab: Es wird schon nach ca. 10s ein Soft-Reset durchgeführt. Das wurde vorher nicht gemacht...oder ich habe es mehrere Male übersehen.

Nun werde ich sukzessive so viel wie nötig wieder hinzufügen und abwarten, ob es nochmal kracht. Danke!

**Reply by SvenLuebke** [reply]
Aber einer noch unbekannten Filegröße wird nach dem Flashen zwischendurch kein Reset mehr ausgeführt. Kurz nach der Ausgabe "TFFS Name Table N" hätte eigentlich ein Soft-Reset stattfinden müssen.
```
[    8.780000] [BUTTON] use GPIO 1 for 'gpio_avm_button_dect'
[    8.790000] [avm_connect][state_machine_init] starting event worker thread
[    8.790000] led_module: Waiting for event system to be ready...
[    9.170000] [0]system-load  54% loadavg 0.24 0.5 0.2 - pgstat: sum=15210 free=8964 slab=1503 alloc=1211/s fault=546/s (sleep 1)
[    9.980000] yaffs: dev is 32505860 name is "mtdblock4" rw
[    9.990000] yaffs: passed flags ""
[   10.480000] [1]system-load  36% loadavg 0.30 0.7 0.2 - task runtime:10% max:cp 9%  pgstat: sum=15132 free=8685 slab=1553 alloc=1259/s fault=631/s (sleep 3)
[   19.170000] [0]system-load  71% loadavg 0.36 0.8 0.3 - task runtime:38% max:cp 35%  pgstat: sum=15144 free=335 slab=1673 alloc=2290/s fault=90/s (sleep 0)
[   29.170000] [0]system-load  64% loadavg 0.61 0.14 0.5 - task runtime:40% max:cp 37%  pgstat: sum=15142 free=366 slab=1679 alloc=1931/s (sleep 0)
[   31.860000] TFFS Name Table N
[   32.440000] yaffs: dev is 32505863 name is "mtdblock7" rw
[   32.440000] yaffs: passed flags ""
[   39.190000] [0]system-load  59% loadavg 0.67 0.17 0.6 - task runtime:6% max:loop0 1%  pgstat: sum=15318 free=573 slab=1868 alloc=2834/s fault=6479/s (sleep 1)
[   40.720000] [1]system-load  52% loadavg 0.70 0.19 0.6 - task runtime:3% max:modload 1%  pgstat: sum=15333 free=520 slab=1875 alloc=7527/s fault=8240/s (sleep 3)
[   41.660000] [module-alloc-by-name] give 0x12000 bytes at 0x810dd000 to module 'fat' (0x83d000 total bytes left)
[   41.700000] [module-alloc-by-name] give 0x4000 bytes at 0x810ef000 to module 'vfat' (0x839000 total bytes left)
[   41.840000] AVM_WATCHDOG: System Init Ueberwachung 240 Sekunden
```
Das passiert bei einer größeren Firmware einfach nicht mehr und der Bootvorgang wird fortgeführt. Dann crasht die Hardware. Ich vermute, dass das bei jeder 7490 passiert.

**Comment by fda77** [comment]
Ich hab keine Ahnung was du da treibst. Bei mir läuft die 7490. Also ist entweder dein Config vom image irgend wo schlecht oder die config auf der box. Um das herauszufinden, wie immer
 - recovery um alles zu löschen (vorher backup machen)
 - minimalimage, dh ausser box so gut wie nichts auswählen

> ich habe nun sicherheitshalber  [..] viele removal Patches aktiviert, 

-.-

**Reply by SvenLuebke** [reply]
Was ich da treibe? Ich versuche einen lauffähigen Stand zu bekommen! Inzwischen habe ich herausgefunden, dass eine freetz-Image-Größe von knapp 46MB noch läuft, 47,1MB funktioniert bei mir nicht mehr. Das ist 100% reproduzierbar, auf meiner Hardware. Kann ja sein, dass es mit meiner 7490 zusammenhängt oder meiner Umgebung.

Wie groß ist Dein Image? Hast Du mit dem aktuellen Stand ein 48MB freetz-Image auf Deiner Hardware getestet?

So sieht die Ausgabe aus, wenn alles korrekt abläuft
```
[   10.160000] yaffs: dev is 32505860 name is "mtdblock4" rw
[   10.160000] yaffs: passed flags ""
[   19.240000] [0]system-load  71% loadavg 0.22 0.5 0.2 - task runtime:34% max:cp 31%  pgstat: sum=16037 free=349 slab=1691 alloc=2260/s fault=117/s (sleep 3)
[   29.300000] [0]system-load  62% loadavg 0.42 0.10 0.3 - task runtime:41% max:cp 37%  pgstat: sum=16034 free=365 slab=1681 alloc=1920/s (sleep 0)
[   29.430000] [1]system-load  25% loadavg 0.42 0.10 0.3 - task runtime:42% max:cp 38%  pgstat: sum=16035 free=334 slab=1681 alloc=4193/s fault=90/s (sleep 0)
[   30.900000] TFFS Name Table N
[   31.230000] set_reboot_status: Soft-Reboot  - UP(31)UTC(31)FW(07.57)HW(185)HWS(6)BV(1.1964)BN(107813)FS(07.57)BT(1)BD(0)
[   31.240000] Res
ROM VER: 1.1.4
CFG 05
** START 

RVEC bf200000
```

**Reply by fda77** [reply]
Ah, du benutzt Sachen von PeterPawn. Ich kenne mich damit nicht aus, frag am besten dort nach wo die Scripte her sind
